### PR TITLE
Add many tests for specialized functions

### PR DESCRIPTION
**Pull request by abelsiqueira**
Due to #113, I removed `HS10` from the tests.

### CODE REVIEW

File: test/test_specialized.jl [hunk 1]
```diff
@@ -295,6 +295,58 @@ function test_specinterface(nlp::CUTEstModel, comp_nlp::AbstractNLPModel)
       result = zeros(J(x0)*v)
       cjprod!(nlp.meta.nvar, nlp.meta.ncon, false, false, x0, ones(nlp.meta.nvar), nlp.meta.nvar, result, nlp.meta.ncon)
 
+      nnzj = cdimsj()
+      Jx = jac(comp_nlp, x0)
+      @fact nnzj --> sum(Jx .!= 0) + nlp.meta.nvar
+
+      Wx = hess(comp_nlp, x0, y=y0)
+      nnzh = cdimsh()
```
Comment (dpo) [inline test/test_specialized.jl]: I would have to look at the code in detail, but would guess you're supposed to call `cdimsh()` before `hess()`.

Comment (abelsiqueira) [inline test/test_specialized.jl]: It's possible, but `CUTEstModel` constructor calls `cdimsh`.

Comment (abelsiqueira) [inline test/test_specialized.jl]: By the way, that `hess` is for the `ADNLPModel`, for comparison with the `CUTEstModel`.

Comment (dpo) [inline test/test_specialized.jl]: Ah, good point. So there's an issue if you call `cdimsh()` several times. That's probably reproducible in Fortran.

**Comment by dpo** [comment]
Woohoo!

**Pull request merged**